python/lsst/meas/extensions/convolved/convolved.py

```diff
@@ -0,0 +1,503 @@
+    pass
+
+
+ApertureFluxConfig = makeConfigClass(lsst.meas.base.ApertureFluxControl)
+
+
+class ConvolvedFluxData(Struct):
+    """A `lsst.pipe.base.Struct` for convolved fluxes
+
+    Attributes
+    ----------
+    deconvKey : `lsst.afw.table.Key_Flag`
+        Key to set flag indicating no measurement was made due to the need to deconvolve
+    aperture : `lsst.meas.base.CircularApertureFluxAlgorithm`
+        Measurement algorithm to perform aperture flux measurements
+    kronKeys : `lsst.pipe.base.Struct`
+        Container for Kron results or `None` if no Kron radius is available; when set,
+        includes `result` (`lsst.meas.base.FluxResultKey`: keys to set results from Kron
+        flux measurement) and `flag` (`lsst.afw.table.Key_Flag`: key to set failure flag
+        for Kron measurement).
+    """
+
+    def __init__(self, name, schema, seeing, config, metadata, doKron):
+        deconvKey = schema.addField(name + "_deconv", type="Flag",
+                                    doc="deconvolution required for seeing %f; no measurement made" %
+                                    (seeing,))
+        aperture = lsst.meas.base.CircularApertureFluxAlgorithm(config.aperture.makeControl(), name,
+                                                                schema, metadata)
+        kronKeys = Struct(
+            result=lsst.meas.base.FluxResultKey.addFields(schema, name + "_kron",
+                                                          doc="convolved Kron flux: seeing %f" % (seeing,)),
+            flag = schema.addField(name + "_kron_flag", type="Flag",
+                                   doc="convolved Kron flux failed: seeing %f" % (seeing,)),
+        ) if doKron else None
+        Struct.__init__(self, deconvKey=deconvKey, aperture=aperture, kronKeys=kronKeys)
+
+
+class ConvolvedFluxConfig(lsst.meas.base.SingleFramePluginConfig):
+    # convolution
+    seeing = ListField(dtype=float, default=[3.5, 5.0, 6.5], doc="list of target seeings (FWHM, pixels)")
+    kernelScale = Field(dtype=float, default=4.0, doc="scaling factor of kernel sigma for kernel size")
+    # aperture flux
+    aperture = ConfigField(dtype=ApertureFluxConfig, doc="Aperture photometry parameters")
+    # Kron flux
+    kronRadiusName = Field(dtype=str, default="ext_photometryKron_KronFlux_radius",
+                           doc="name of Kron radius field in reference")
+    maxSincRadius = Field(dtype=float, default=10.0,
+                          doc="Largest aperture for which to use the sinc aperture code for Kron (pixels)")
+    kronRadiusForFlux = Field(dtype=float, default=2.5, doc="Number of Kron radii for Kron flux")
+
+    def setDefaults(self):
+        lsst.meas.base.SingleFramePluginConfig.setDefaults(self)
+        # Don't need the full set of apertures because the larger ones aren't affected by the convolution
+        self.aperture.radii = [3.3, 4.5, 6.0]
+
+    def getBaseNameForSeeing(self, seeing, name=PLUGIN_NAME):
```

Comment:
They don't need the plugin, but they do need the config: if you set the config, then you can determine the names. That suggests that they should be methods on the config.  I guess they could also be free functions that take the config as a parameter, but I like putting them in the config because it makes them easier to call, and they really are functions of the config.